include/oneapi/dpl/pstl/hetero/numeric_impl_hetero.h

```diff
@@ -91,23 +102,249 @@ __pattern_transform_reduce(_ExecutionPolicy&& __exec, _ForwardIterator __first,
+
+            if (__is_full_group)
+                __single_group_scan<_Inclusive::value, __num_elems_per_item, __wg_size,
+                                    /* _IsFullGroup= */ true>::__launch_static_bounds_scan(__exec, __in_rng.all_view(),
+                                                                                           __out_rng.all_view(), __n,
+                                                                                           __init, __binary_op,
+                                                                                           __unary_op);
+            else
+                __single_group_scan<_Inclusive::value, __num_elems_per_item, __wg_size,
+                                    /* _IsFullGroup= */ false>::__launch_static_bounds_scan(__exec, __in_rng.all_view(),
+                                                                                            __out_rng.all_view(), __n,
+                                                                                            __init, __binary_op,
+                                                                                            __unary_op);
+        };
+        if (__n <= 16)
+            __single_group_scan_f(std::integral_constant<::std::uint16_t, 16>{});
+        else if (__n <= 32)
+            __single_group_scan_f(std::integral_constant<::std::uint16_t, 32>{});
+        else if (__n <= 64)
+            __single_group_scan_f(std::integral_constant<::std::uint16_t, 64>{});
+        else if (__n <= 128)
+            __single_group_scan_f(std::integral_constant<::std::uint16_t, 128>{});
+        else if (__n <= 256)
+            __single_group_scan_f(std::integral_constant<::std::uint16_t, 256>{});
+        else if (__n <= 512)
+            __single_group_scan_f(std::integral_constant<::std::uint16_t, 512>{});
+        else if (__n <= 1024)
+            __single_group_scan_f(std::integral_constant<::std::uint16_t, 1024>{});
+        else if (__n <= 2048)
+            __single_group_scan_f(std::integral_constant<::std::uint16_t, 2048>{});
+        else if (__n <= 4096)
+            __single_group_scan_f(std::integral_constant<::std::uint16_t, 4096>{});
+        else if (__n <= 8192)
+            __single_group_scan_f(std::integral_constant<::std::uint16_t, 8192>{});
+        else
+            __single_group_scan_f(std::integral_constant<::std::uint16_t, 16384>{});
+    }
+    else
+    {
+        __single_group_scan<_Inclusive::value>::template __launch_dynamic_bounds_scan<_DynamicGroupScanKernel>(
+            __exec, __in_rng.all_view(), __out_rng.all_view(), __n, __init, __binary_op, __unary_op, __max_wg_size);
+    }
+}
+
+template <typename _ExecutionPolicy, typename _InRng, typename _OutRng, typename _UnaryOperation, typename _InitType,
+          typename _BinaryOperation, typename _Inclusive>
+void
+__pattern_transform_scan_multi_group(_ExecutionPolicy&& __exec, _InRng&& __in_rng, _OutRng __out_rng,
```

Comment:
That pattern should be moved to following file:
https://github.com/oneapi-src/oneDPL/blob/6cd24948245a409eb6c2f61c2d07af43471f0a43/include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl.h

Reply:
Fixed. It's now in parallel_backend_sycl.h.